demisto_sdk/commands/common/content/objects/pack_objects/integration/integration.py

```diff
@@ -19,3 +23,23 @@ def png_path(self) -> Optional[Path]:
     def description_path(self) -> Optional[Path]:
         patterns = [f"{self.path.stem}_description.md"]
         return next(self._path.parent.glob(patterns=patterns), None)
+
+    def upload(self, client: demisto_client = None):
+        """
+        Upload the integration to demisto_client
+        Args:
+            client: The demisto_client object of the desired XSOAR machine to upload to.
+
+        Returns:
+            The result of the upload command from demisto_client
+        """
+        if self.is_unify():
+            return client.integration_upload(file=self.path)  # type: ignore
+        else:
+            with tempfile.TemporaryDirectory() as dir:
+                unified_files = self._unify(dir)
+                for file in unified_files:
+                    if (str(file)[-7:] == '_45.yml') == (get_demisto_version(client) < parse('4.6.0')):
```

Comment:
shouldn't this apply to all file types?

Reply:
The `self._unify` method creates 2 files (if dockerimage45 is present) so I want to upload the relevant one. For other file types the fromversion, toversion should suffice and that's what it's doing